python-threatexchange/threatexchange/exchanges/impl/file_api.py

```diff
@@ -47,6 +47,7 @@ class FileCollaborationConfig(
     )
 
 
+@dataclass
```

Comment:
nit: just to get the automatic init method?

Is there anything else 'dataclassey' about this class?

Reply:
The main thing dataclass gets you is an init method :P  'dataclassey' and "data class" don't have any formal definition to my knowledge, in which case one way we could define it is simply by what the dataclass decorator does as side effects, in which case automatic `__init__` seems like a dataclassey outcome.

Reply:
It also adds hash and eq methods and the ability to freeze the object. Which useful for objects that are serialized or passed along as data -- sans behavior.

But, I'm probably reading in too much from kotlin dataclasses which has a more formal defn.